src/client/components/AddNewGroupComponent/GroupHandler.js

```diff
@@ -0,0 +1,38 @@
+import React, { useState } from 'react';
+import { Route, Switch, BrowserRouter } from 'react-router-dom';
+import AddNewGroup from './AddNewGroup';
+
+const GroupHandler = () => {
+  const [newGroup, setNewGroup] = useState('');
+  const [addGroup, setaddGroup] = useState('');
+
+  const AddGroupName = (groupName) => {
+    setNewGroup(groupName.target.value);
+  };
+
+  const groupdHandler = () => {
+    setaddGroup(newGroup);
+    setNewGroup('');
+  };
+  return (
+    <>
+      <BrowserRouter>
+        <Switch>
+          <Route
+            exact
+            path="/channel"
+            render={() => (
+              <AddNewGroup
+                AddGroupName={(groupName) => AddGroupName(groupName)}
```

Comment:
Parameter should be named `e` in this case since it is not actually a group name but an event itself.
Or change the event to actually send only the text in which case it will become easier to understand the code.